src/usr/hdat/hdatiplparms.C

```diff
@@ -724,18 +725,97 @@ void HdatIplParms::hdatGetSystemParamters()
 
     this->iv_hdatIPLParams->iv_sysParms.hdatSplitCoreMode = 1;
 
-    TARGETING::ATTR_SYSTEM_BRAND_NAME_type l_systemBrandName = {0};
-    if(l_pSysTarget->tryGetAttr<TARGETING::ATTR_SYSTEM_BRAND_NAME>
-                                                         (l_systemBrandName))
+    /* Read UTIL::F5 keyword (maxlen=16)*/
+    errlHndl_t l_err = NULL;
+    size_t  dataSize = 0;
+    char    dataBuffer[17]={0};
+    char    sysVendorname[64]={0};
+    Target* node_tgt = UTIL::getCurrentNodeTarget();
+
+    l_err = deviceRead(node_tgt, NULL, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F5 ));
+
+    if (l_err || (dataSize > sizeof(dataBuffer) - 1))
```

Comment:
(dataSize+1 > sizeof(dataBuffer) )
   is more safe for arbitrary sizes of dataBuffer

Reply:
done